toolchain/check/name_lookup.cpp

```diff
@@ -101,6 +102,30 @@ auto LookupUnqualifiedName(Context& context, Parse::NodeId node_id,
                                             .specific_id = specific_id},
                                 /*required=*/false);
         non_lexical_result.scope_result.is_found()) {
+      // In an interface definition, replace associated entity `M` with
+      // `Self.M` (where the `Self` is the `Self` of the interface).
+      const auto& scope = context.name_scopes().Get(lookup_scope_id);
+      if (scope.is_interface_definition()) {
+        SemIR::InstId target_inst_id =
+            non_lexical_result.scope_result.target_inst_id();
+        if (auto inst = context.insts().TryGetAs<SemIR::AssociatedEntity>(
```

Comment:
This could also be an alias to an associated entity or an import ref or something like that. I think it'd be best to either look at the type of the instruction or its constant value.